src/main/java/org/jboss/tools/intellij/openshift/ui/project/CreateNewProjectDialog.java

```diff
@@ -33,84 +32,84 @@
+    this.kind = kind;
+    this.allProjects = allProjects;
+    init();
+  }
 
-    @Override
-    protected void init() {
-        super.init();
-        setOKButtonText("Create");
-        setTitle("Create New " + kind);
-    }
+  @Override
+  protected void init() {
+    super.init();
+    setOKButtonText("Create");
+    setTitle("Create New " + kind);
+  }
 
-    @Override
-    protected @Nullable JComponent createCenterPanel() {
-        JComponent panel = new JPanel(new MigLayout(
-          "flowx, ins 0, gap 0, fillx, filly, hidemode 3",
-          "[left]10[" + WIDTH +",fill]"));
-        JLabel newActiveProjectLabel = new JBLabel("New " + kind.toLowerCase() + ":", SwingConstants.LEFT);
-        newActiveProjectLabel.setBorder(JBUI.Borders.empty(10, 0));
-        panel.add(newActiveProjectLabel, "left, bottom");
-        this.newProjectTextField = new JBTextField();
-        newProjectTextField.selectAll();
-        panel.add(newProjectTextField, "pushx, growx, wrap");
-        ComponentValidator activeProjectValidator = new ComponentValidator(myDisposable)
-          .withValidator(new ActiveProjectValidator())
-          .installOn(newProjectTextField)
-          .andRegisterOnDocumentListener(newProjectTextField);
-        activeProjectValidator.revalidate();
-        return panel;
-    }
+  @Override
+  protected @Nullable JComponent createCenterPanel() {
+    JComponent panel = new JPanel(new MigLayout(
+      "flowx, ins 0, gap 0, fillx, filly, hidemode 3",
+      "[left]10[" + WIDTH + ",fill]"));
+    JLabel newActiveProjectLabel = new JBLabel("New " + kind.toLowerCase() + ":", SwingConstants.LEFT);
+    newActiveProjectLabel.setBorder(JBUI.Borders.empty(10, 0));
+    panel.add(newActiveProjectLabel, "left, bottom");
+    this.newProjectTextField = new JBTextField();
+    newProjectTextField.selectAll();
+    panel.add(newProjectTextField, "pushx, growx, wrap");
+    ComponentValidator activeProjectValidator = new ComponentValidator(myDisposable)
+      .withValidator(new ActiveProjectValidator())
+      .installOn(newProjectTextField)
+      .andRegisterOnDocumentListener(newProjectTextField);
+    activeProjectValidator.revalidate();
+    return panel;
+  }
 
-    @Override
-    protected void doOKAction() {
-        super.doOKAction();
-        this.newProject = newProjectTextField.getText();
-    }
+  @Override
+  protected void doOKAction() {
+    super.doOKAction();
+    this.newProject = newProjectTextField.getText();
+  }
 
-    public String getNewProject() {
-        return newProject;
-    }
+  public String getNewProject() {
+    return newProject;
+  }
 
-    private class ActiveProjectValidator implements Supplier<ValidationInfo> {
+  private class ActiveProjectValidator implements Supplier<ValidationInfo> {
 
-        private final Pattern nameRegex = Pattern.compile(
-          "^[a-z0-9]+(([.-])?[a-z0-9])$");
+    private final Pattern nameRegex = Pattern.compile(
+      "^[a-z0-9]+(?:-[a-z0-9]+){0,64}$");
```

Comment:
Reading https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names I believe that 2 dashes are legal but we're not allowing it
<img width="445" alt="image" src="https://github.com/user-attachments/assets/cc3251f2-7ffc-4a54-a9b4-48ef812bb1ef">

Reply:
fixed